### DISCUSSION TITLE
Could you recommend me some parameters or flow variables?

### DISCUSSION THREAD

**Discussion by huang-kaixin** [The-OpenROAD-Project/OpenROAD-flow-scripts#1565] [Q&A]
I'm researching RTL to GDS automatic parameter tuning, like autotuner. I'd like to know about parameters or flow variables that have a significant impact on metrics such as route__wirelength, power__total, design__instance__area, and routability. Thanks for your responses.@maliberty

**Comment by rovinski** [comment]
Most of the flow variables are somewhat documented on the [docs page](https://openroad-flow-scripts.readthedocs.io/en/latest/user/FlowVariables.html). Pretty much any variable you set has implications downstream in the flow. I don't know if it's possible to be more specific than that - it's a massive, hyperdimensional optimization space.

**Reply by huang-kaixin** [reply]
Thanks for your response. I do understand it's a massive, hyperdimensional optimization space. However, some flow variables within it have a relatively minor impact on the final metrics, while others have a significant impact. I'm trying to identify variables that have a substantial influence on the ultimate metrics. In other words, I'm narrowing down the optimization space to find the most accurate variables.

**Reply by rovinski** [reply]
Some variables may only have a minor impact because the region around the default value has a shallow gradient. If you change that default to some other area, there may be a much steeper gradient.

To add to that, the optimal values / search space / gradients / etc. change drastically depending on both the technology _and_ design. If you are looking at a specific design for a specific technology, then that can help to narrow things down a bit, but otherwise it's nearly impossible to make generalized statements about which parameters have the most impact. 

The best variables to look at are the ones which are numerical values rather than strings. There honestly aren't that many of them exposed in ORFS. There are many more under the hood in OR.

Take a look at an example [autotuner.json here](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/6c4437edabdaddde9f100d76cc77b568c6cc073d/flow/designs/asap7/aes/autotuner.json)

**Reply by maliberty** [reply]
In the autotuner we played with some obvious ones.  Depending on your goal and your capacity there are tons more.
